src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -47,7 +61,12 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
         const buttonParams: ButtonsOptions = {
             style: buttonStyle,
             onApprove: data => this._tokenizePayment(data, submitForm),
-            onClick: async (_, actions) => onValidate(actions.resolve, actions.reject),
+            onClick: async (_, actions) => {
+                this._initializePollingMechanism(submitForm, gatewayId);
+
+                return onValidate(actions.resolve, actions.reject);
+            },
+            onCancel: () => this._deinitializePollingMechanism(),
```

Comment:
Do you need to cancel the polling if `onApprove` is called?

Reply:
Good point, thanks

Reply:
fixed